pkg/azure/provider/provider_test.go

```diff
@@ -0,0 +1,1180 @@
+	g.Expect(err).To(BeNil())
+	diskAccess, err := factory.NewDiskAccessBuilder().WithClusterState(clusterState).WithAPIBehaviorSpec(diskAccessAPIBehaviorSpec).Build()
+	g.Expect(err).To(BeNil())
+	factory.
+		WithVirtualMachineAccess(vmAccess).
+		WithResourceGroupsAccess(rgAccess).
+		WithNetworkInterfacesAccess(nicAccess).
+		WithDisksAccess(diskAccess)
+
+	return factory
+}
+
+func createDefaultFakeFactoryForCreateMachine(g *WithT, clusterState *fakes.ClusterState) *fakes.Factory {
+	return createFakeFactoryForCreateMachineWithAPIBehaviorSpecs(g, clusterState.ProviderSpec.ResourceGroup, clusterState, nil, nil, nil, nil, nil)
+}
+
+func createFakeFactoryForCreateMachineWithAPIBehaviorSpecs(g *WithT, resourceGroup string, clusterState *fakes.ClusterState,
+	vmAccessAPIBehaviorSpec *fakes.APIBehaviorSpec,
+	subnetAccessAPIBehaviorSpec *fakes.APIBehaviorSpec,
+	nicAccessAPIBehaviorSpec *fakes.APIBehaviorSpec,
+	vmImageAccessAPIBehaviorSpec *fakes.APIBehaviorSpec,
+	mktPlaceAgreementAccessAPIBehaviorSpec *fakes.APIBehaviorSpec) *fakes.Factory {
+
+	factory := fakes.NewFactory(resourceGroup)
+	vmAccess, err := factory.NewVirtualMachineAccessBuilder().WithClusterState(clusterState).WithAPIBehaviorSpec(vmAccessAPIBehaviorSpec).Build()
+	g.Expect(err).To(BeNil())
+	vmImageAccess, err := factory.NewImageAccessBuilder().WithClusterState(clusterState).WithAPIBehaviorSpec(vmImageAccessAPIBehaviorSpec).Build()
+	g.Expect(err).To(BeNil())
+	subnetAccess, err := factory.NewSubnetAccessBuilder().WithClusterState(clusterState).WithAPIBehaviorSpec(subnetAccessAPIBehaviorSpec).Build()
+	g.Expect(err).To(BeNil())
+	mktPlaceAgreementAccess, err := factory.NewMarketPlaceAgreementAccessBuilder().WithClusterState(clusterState).WithAPIBehaviorSpec(mktPlaceAgreementAccessAPIBehaviorSpec).Build()
+	g.Expect(err).To(BeNil())
+	nicAccess, err := factory.NewNICAccessBuilder().WithClusterState(clusterState).WithAPIBehaviorSpec(nicAccessAPIBehaviorSpec).Build()
+	g.Expect(err).To(BeNil())
+	diskAccess, err := factory.NewDiskAccessBuilder().WithClusterState(clusterState).Build()
+	g.Expect(err).To(BeNil())
+	factory.
+		WithVirtualMachineAccess(vmAccess).
+		WithVirtualMachineImagesAccess(vmImageAccess).
+		WithSubnetAccess(subnetAccess).
+		WithMarketPlaceAgreementsAccess(mktPlaceAgreementAccess).
+		WithNetworkInterfacesAccess(nicAccess).
+		WithDisksAccess(diskAccess)
+
+	return factory
+}
+
+func getVMNamesFromListMachineResponse(response *driver.ListMachinesResponse) []string {
+	if response == nil {
+		return []string{}
+	}
+	vmNames := make([]string, 0, len(response.MachineList))
+	for _, vmName := range response.MachineList {
+		vmNames = append(vmNames, vmName)
+	}
+	return vmNames
+}
+
+func checkAndGetWrapperAzResponseError(g *WithT, err error, expectedStatusCode codes.Code) *azcore.ResponseError {
+	var statusErr *status.Status
+	g.Expect(errors.As(err, &statusErr)).To(BeTrue())
+	g.Expect(statusErr.Code(), expectedStatusCode)
```

Comment:
two suggestions:
* could you also assert the statusErr.Message 
* could you extract this MCM error assertion to a new method. This way it would easier to follow the flow at the caller side also

Reply:
Asserting the message is not wise, since that is a formed sentence. It is always good to check the error code (if present), type of error, essentially immutable/constants or types.

Reply:
did not quite understand the second suggestion.